src/modules/uORB/uORBDeviceNode.cpp

```diff
@@ -782,3 +786,37 @@ int uORB::DeviceNode::update_queue_size(unsigned int queue_size)
 	_queue_size = queue_size;
 	return PX4_OK;
 }
+
+bool uORB::DeviceNode::register_work_item(px4::WorkItem *item)
+{
+	if (item != nullptr) {
+		_registered_work_items.add(item);
+		_work_items_available = true;
```

Comment:
Shouldn't `List` have an `empty()` or `size()` method instead of this additional bool to keep track of the state?

Reply:
Yes.

Reply:
Actually, it was done to be lazy in the orb publish call above.

Reply:
I've added an `empty()` method.